packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -0,0 +1,283 @@
+import appscaling = require('@aws-cdk/aws-applicationautoscaling');
+import cloudwatch = require('@aws-cdk/aws-cloudwatch');
+import ec2 = require('@aws-cdk/aws-ec2');
+import elbv2 = require('@aws-cdk/aws-elasticloadbalancingv2');
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { BaseTaskDefinition, NetworkMode } from '../base/base-task-definition';
+import { cloudformation } from '../ecs.generated';
+import { ScalableTaskCount } from './scalable-task-count';
+
+/**
+ * Basic service properties
+ */
+export interface BaseServiceProps {
+  /**
+   * Number of desired copies of running tasks
+   *
+   * @default 1
+   */
+  desiredCount?: number;
+
+  /**
+   * A name for the service.
+   *
+   * @default CloudFormation-generated name
+   */
+  serviceName?: string;
+
+  /**
+   * The maximum number of tasks, specified as a percentage of the Amazon ECS
+   * service's DesiredCount value, that can run in a service during a
+   * deployment.
+   *
+   * @default 200
+   */
+  maximumPercent?: number;
+
+  /**
+   * The minimum number of tasks, specified as a percentage of
+   * the Amazon ECS service's DesiredCount value, that must
+   * continue to run and remain healthy during a deployment.
+   *
+   * @default 50
+   */
+  minimumHealthyPercent?: number;
+
+  /**
+   * Time after startup to ignore unhealthy load balancer checks.
+   *
+   * @default ??? FIXME
+   */
+  healthCheckGracePeriodSeconds?: number;
+
+  /**
+   * Fargate platform version to run this service on
+   *
+   * Unless you have specific compatibility requirements, you don't need to
+   * specify this.
+   *
+   * @default Latest
+   */
+  platformVersion?: FargatePlatformVersion;
+}
+
+/**
+ * Base class for Ecs and Fargate services
+ */
+export abstract class BaseService extends cdk.Construct
+  implements elbv2.IApplicationLoadBalancerTarget, elbv2.INetworkLoadBalancerTarget, cdk.IDependable {
+
+  /**
+   * CloudFormation resources generated by this service
+   */
+  public readonly dependencyElements: cdk.IDependable[];
+
+  /**
+   * Manage allowed network traffic for this construct
+   */
+  public abstract readonly connections: ec2.Connections;
+
+  /**
+   * ARN of this service
+   */
+  public readonly serviceArn: string;
+
+  /**
+   * Name of this service
+   */
+  public readonly serviceName: string;
+
+  /**
+   * Name of this service's cluster
+   */
+  public readonly clusterName: string;
+
+  protected loadBalancers = new Array<cloudformation.ServiceResource.LoadBalancerProperty>();
+  protected networkConfiguration?: cloudformation.ServiceResource.NetworkConfigurationProperty;
+  protected readonly abstract taskDef: BaseTaskDefinition;
```

Comment:
Since this has to be either a concrete `EcsTaskDefinition` or `FargateTaskDefinition`, I believe this is set in the constructors for `EcsService` and `FargateService`, respectively.